sites/svelte-5-preview/src/routes/docs/content/03-appendix/02-breaking-changes.md

```diff
@@ -285,3 +285,37 @@ Note that whereas Svelte 4 would treat `<svelte:element this="input">` (for exam
 ### `mount` plays transitions by default
 
 The `mount` function used to render a component tree plays transitions by default unless the `intro` option is set to `false`. This is different from legacy class components which, when manually instantiated, didn't play transitions by default.
+
+### `<img src={...}>` and `{@html ...}` hydration mismatches are not repaired
+
+In Svelte 4, when the value of an `<img src={...}>` attribute or a `{@html ...}` tag were different between the server and the client (i.e. there was a hydration mismatch), that mismatch was repaired. This is very costly though, and therefore is no longer done in Svelte 5. Instead, you'll now get a warning at DEV time when a mismatch is detected. If you know the value can be different between server and client, you can do something like this:
+
+```svelte
+<script>
+	import { onMount, tick } from 'svelte';
+
+	let raw_html = $state(
+		typeof window === 'undefined'
+			? '<h1>Server</h1>'
+			: '<h1>Client</h1>'
+	);
+	let src = $state(
+		typeof window === 'undefined'
+			? '/server.jpg'
+			: '/client.jpg'
+	);
+
+	onMount(async () => {
+		// force update
+		raw_html = '';
+		src = '';
+		await tick();
+		// set client values
+		raw_html = '<h1>Client</h1>';
+		src = '/client.jpg';
+	});
+</script>
+
+{@html raw_html}
+<img {src} />
```

Comment:
another way we could tackle this, which is arguably a better practice since it scales to an arbitrary number of variables:

```suggestion
  let mounted = $state(false);
	$effect(() => {
		// force `markup` and `src` to be re-evaluated
		mounted = true;
	});
</script>

{@html (mounted, markup)}
<img src={(mounted, src)} />
```

Reply:
This doesn't work for `@html`, because it checks against the previous value and does nothing if it's the same

Reply:
Hmm. I realised it doesn't really any sense anyway — if you're dealing with local state, you should just do this:

```svelte
<script>
	import { onMount, tick } from 'svelte';

	let markup = $state('<h1>Server</h1>');
	let src = $state('/server.jpg');

	onMount(() => {
		markup = '<h1>Client</h1>';
		src = '/client.jpg';
	});
</script>
```

So this only makes sense if you're dealing with props, in which case you don't _know_ what values to set to. Something like this would make more sense:

```svelte
<script>
	let { markup, src } = $props();

	if (typeof window !== 'undefined') {
		// stash the values...
		const initial = { markup, src };

		// unset them...
		markup = src = undefined;

		$effect(() => {
			// ...and reset after we've mounted
			markup = initial.markup;
			src = initial.src;
		});
	}
</script>

{@html markup}
<img {src} />
```